activerecord/lib/active_record/fixtures.rb

```diff
@@ -975,20 +975,12 @@ def setup_fixtures(config = ActiveRecord::Base)
 
         # When connections are established in the future, begin a transaction too
         @connection_subscriber = ActiveSupport::Notifications.subscribe("!connection.active_record") do |_, _, _, _, payload|
-          spec_name = payload[:spec_name] if payload.key?(:spec_name)
+          connection = payload[:connection] if payload.key?(:connection)
 
-          if spec_name
-            begin
-              connection = ActiveRecord::Base.connection_handler.retrieve_connection(spec_name)
-            rescue ConnectionNotEstablished
-              connection = nil
-            end
-
-            if connection && !@fixture_connections.include?(connection)
-              connection.begin_transaction joinable: false
-              connection.pool.lock_thread = true
-              @fixture_connections << connection
-            end
+          if connection && !@fixture_connections.include?(connection)
+            connection.begin_transaction joinable: false
+            connection.pool.lock_thread = true
```

Comment:
This seems dangerous: it could lead to us locking on any random thread's connection. (Actually already true, now that I read it.. but the change to *when* this occurs seems to make it much more likely to misbehave.)